lib/src/repository/repository_url_parser.dart

```diff
@@ -40,18 +43,19 @@ Repository? tryParseRepositoryUrl(String input) {
 
   // Normalizing the URL path and rejecting URLs that may differ more than
   // a trailing slash.
-  final normalizedUri = Uri.tryParse(p.normalize(uri.path));
-  if (normalizedUri == null) {
-    return null;
-  }
+  final normalizedUri = Uri.parse(p.normalize(uri.path));
   if (uri.path != normalizedUri.path && uri.path != '${normalizedUri.path}/') {
-    return null;
+    throw FormatException(
+        'URL path is not normalized: `${uri.path}` != `${normalizedUri.path}`');
   }
   // detect repo vs path segments
   final segments =
       normalizedUri.pathSegments.where((s) => s.isNotEmpty).toList();
   final repoSegmentIndex = _repoSegmentIndex(provider, segments);
-  if (repoSegmentIndex < 0) return null;
+  if (repoSegmentIndex < 0) {
+    throw FormatException(
+        'Could not find repository segment separator in `${normalizedUri.path}`.');
```

Comment:
I somehow still think this message is hard to grog. I wouldn't know what a "repository segment separator" is.

We should probably document how we expect the repository link be formattet.

Maybe we should link to https://dart.dev/tools/pub/pubspec#repository and extend that with an explanation of how to deep link into the repo.

Reply:
I've updated with the message to report the issue at `pana`'s issue tracker. I think ultimately we want to have unrecognized examples reported, and keep the officially supported description vague. At this point in the parsing it will probably not fail at github deep links, but may fail on niche git repository structures like self-hosted git frontends.